### ISSUE TITLE
A formal description of string interpolation seems to be missing from the docs

### ISSUE THREAD

**Issue by azazel75** [elixir-lang/elixir#9350]
Hello, 

I'm an elixir newbie and today I was looking to a more comprehensive description of how interpolation works, i.e. what's possible to put inside `#{}` sections. 

There's a brief mention of it in the getting started: https://elixir-lang.org/getting-started/basic-types.html#strings but then I moved to the reference, trying the [String datatype documentation](https://hexdocs.pm/elixir/1.9.1/String.html#content) but without luck

It would be nice to have the following informations:
* formatting for various kind of data, i.e. how many decimals for numbers, padding, alignement;
* if expressions are allowed;
* if the syntax is recursive, i.e. if expressions are allowed, the syntax permits the opening a new double quoted string inside of it, maybe with its own interpolations inside?
* Is it possible to dynamically prepare the string to be interpolated (a kind of templating)? and if not, where should I look for?

In summary, I was looking similar to https://docs.python.org/3/reference/lexical_analysis.html#formatted-string-literals

Anyway, it's possible that I was just unable to find it. If that's the case I'm sorry fort that.
